paasta_tools/utils.py

```diff
@@ -3076,20 +3078,28 @@ def read_service_instance_names(
     service: str, instance_type: str, cluster: str, soa_dir: str
 ) -> Collection[Tuple[str, str]]:
     instance_list = []
-    conf_file = f"{instance_type}-{cluster}"
+    if instance_type == "spark":
```

Comment:
we should probably add a comment here as well since this is going to look pretty surprising otherwise